swarm/pss/pss_test.go

```diff
@@ -482,16 +481,31 @@ func testAsymSend(t *testing.T) {
 	}
 }
 
+type Job struct {
+	Msg      []byte
+	SendNode discover.NodeID
+	RecvNode discover.NodeID
+}
+
+func worker(id int, jobs <-chan Job, rpcs map[discover.NodeID]*rpc.Client, pubkeys map[discover.NodeID][]byte, hextopic string) {
+	for j := range jobs {
+		rpcs[j.SendNode].Call(nil, "pss_sendAsym", common.ToHex(pubkeys[j.RecvNode]), hextopic, j.Msg)
+	}
+}
+
 // params in run name:
 // nodes/msgs/addrbytes/adaptertype
 // if adaptertype is exec uses execadapter, simadapter otherwise
 func TestNetwork(t *testing.T) {
-	//t.Run("8/1024/2/exec", testNetwork)
-	t.Run("256/128/4/sock", testNetwork)
+	t.Run("3/2000/4/sock", testNetwork)
+	t.Run("4/2000/4/sock", testNetwork)
+	t.Run("8/2000/4/sock", testNetwork)
+	t.Run("16/2000/4/sock", testNetwork)
```

Comment:
what's the reason for having three whole tests for n<10? And why is the last one a `sim`?

Reply:
While performing the tests when the code was still deadlocking, I noticed that it would deadlock at different thresholds, depending on the number of nodes in the system - probably to do with the concurrency levels being different if you have fewer nodes vs many nodes.

Ultimately there is no difference now, but I still kept the tests around.

Reply:
The last test is a `sim` just so that we make sure that tests are passing also with a synchronous adapter, rather than only with asynchronous (after all we have a big buffer on `sock` by default).

It would probably be good to add `tcp` and `exec` adapters as well, but I just don't want to make these tests too heavy. If `sim` and `sock` are passing, we have very high confidence that the code is correct.

Reply:
Also note that "3/2000/4/sock" is different to "16/2000/4/sock" as in both tests we send a total of 2000 messages, so in the first test each of the nodes actually processes more messages on average (2000/3) compared to the other test (2000/16). Obviously there is a random element involved, but you get the idea.

Reply:
Yes, good point!